[34m[1mwandb[39m[22m: [33mWARNING[39m Calling run.save without any arguments is deprecated.Changes to attributes are automatically persisted.
I0127 20:41:31.199011 140626491549568 run.py:115] Using CLRS21 spec: {'train': {'num_samples': 1000, 'length': 16, 'seed': 5}, 'val': {'num_samples': 32, 'length': 16, 'seed': 34}, 'test': {'num_samples': 32, 'length': 64, 'seed': 41}}
I0127 20:41:34.512405 140626491549568 xla_bridge.py:231] Unable to initialize backend 'tpu_driver': NOT_FOUND: Unable to find driver in registry given worker:
2022-01-27 20:41:34.526302: E external/org_tensorflow/tensorflow/stream_executor/cuda/cuda_driver.cc:271] failed call to cuInit: CUDA_ERROR_NO_DEVICE: no CUDA-capable device is detected
I0127 20:41:34.526612 140626491549568 xla_bridge.py:231] Unable to initialize backend 'gpu': FAILED_PRECONDITION: No visible GPU devices.
I0127 20:41:34.527194 140626491549568 xla_bridge.py:231] Unable to initialize backend 'tpu': INVALID_ARGUMENT: TpuPlatform is not available.
W0127 20:41:34.527365 140626491549568 xla_bridge.py:235] No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
I0127 20:42:09.088567 140626491549568 run.py:186] Compiled feedback step in 34.579993 s.
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
adj shape >>  (32, 20, 20)
max
z shape >>  (32, 20, 12)
nxt_hidden shape >>  (32, 20, 12)
I0127 20:42:16.059246 140626491549568 run.py:199] (train) step 0: {'average_accuracy': 0.5010712377899879, 'average_shift': 5.40625}
I0127 20:42:21.580508 140626491549568 run.py:206] (val) step 0: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:42:21.581268 140626491549568 run.py:212] Saving new checkpoint for average accuracy...
I0127 20:42:21.587992 140626491549568 run.py:217] Saving new checkpoint for average shift...
I0127 20:42:32.725832 140626491549568 run.py:199] (train) step 10: {'average_accuracy': 0.5635055071938297, 'average_shift': 4.6875}
I0127 20:42:32.756330 140626491549568 run.py:206] (val) step 10: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:42:44.128365 140626491549568 run.py:199] (train) step 20: {'average_accuracy': 0.5129841254913816, 'average_shift': 5.375}
I0127 20:42:44.158768 140626491549568 run.py:206] (val) step 20: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:42:55.335427 140626491549568 run.py:199] (train) step 30: {'average_accuracy': 0.5245469295174211, 'average_shift': 5.5}
I0127 20:42:55.365278 140626491549568 run.py:206] (val) step 30: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:43:06.562268 140626491549568 run.py:199] (train) step 40: {'average_accuracy': 0.5473919801174058, 'average_shift': 5.0625}
I0127 20:43:06.593065 140626491549568 run.py:206] (val) step 40: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:43:17.781342 140626491549568 run.py:199] (train) step 50: {'average_accuracy': 0.5260001363982594, 'average_shift': 5.75}
I0127 20:43:17.815283 140626491549568 run.py:206] (val) step 50: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:43:34.876646 140626491549568 run.py:199] (train) step 60: {'average_accuracy': 0.4031152594341933, 'average_shift': 1.375}
I0127 20:43:34.938489 140626491549568 run.py:206] (val) step 60: {'average_accuracy': 0.43532592225026434, 'average_shift': 1.0}
I0127 20:43:34.949537 140626491549568 run.py:217] Saving new checkpoint for average shift...
I0127 20:44:03.013780 140626491549568 run.py:199] (train) step 70: {'average_accuracy': 0.5036249363757588, 'average_shift': 5.4375}
I0127 20:44:03.127947 140626491549568 run.py:206] (val) step 70: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:44:30.491281 140626491549568 run.py:199] (train) step 80: {'average_accuracy': 0.503732007962368, 'average_shift': 6.125}
I0127 20:44:30.546228 140626491549568 run.py:206] (val) step 80: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
I0127 20:44:51.180533 140626491549568 run.py:199] (train) step 90: {'average_accuracy': 0.43076110470411944, 'average_shift': 3.375}
I0127 20:44:51.211607 140626491549568 run.py:206] (val) step 90: {'average_accuracy': 0.45875121315088413, 'average_shift': 3.59375}
I0127 20:45:02.756582 140626491549568 run.py:199] (train) step 100: {'average_accuracy': 0.5408656336300008, 'average_shift': 5.3125}
I0127 20:45:02.785292 140626491549568 run.py:206] (val) step 100: {'average_accuracy': 0.5874371744848718, 'average_shift': 4.3125}
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3.7/linecache.py", line 47, in getlines
    return updatecache(filename, module_globals)
  File "/usr/lib/python3.7/linecache.py", line 95, in updatecache
    stat = os.stat(fullname)
KeyboardInterrupt
Original exception was:
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/content/drive/MyDrive/Final Project/new_clrs/clrs/examples/run.py", line 294, in <module>
    app.run(main)
  File "/usr/local/lib/python3.7/dist-packages/absl/app.py", line 312, in run
    _run_main(main, args)
  File "/usr/local/lib/python3.7/dist-packages/absl/app.py", line 258, in _run_main
    sys.exit(main(argv))
  File "/content/drive/MyDrive/Final Project/new_clrs/clrs/examples/run.py", line 183, in main
    cur_loss = model.feedback(feedback)
  File "/content/drive/MyDrive/Final Project/new_clrs/clrs/_src/baselines.py", line 861, in feedback
    self.opt_state,
  File "/content/drive/MyDrive/Final Project/new_clrs/clrs/_src/baselines.py", line 1009, in update
  File "/usr/local/lib/python3.7/dist-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/jax/_src/api.py", line 993, in value_and_grad_f
    ans, vjp_py = _vjp(f_partial, *dyn_args, reduce_axes=reduce_axes)
  File "/usr/local/lib/python3.7/dist-packages/jax/_src/api.py", line 2313, in _vjp
    flat_fun, primals_flat, reduce_axes=reduce_axes)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/ad.py", line 116, in vjp
    out_primals, pvals, jaxpr, consts = linearize(traceable, *primals)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/ad.py", line 103, in linearize
    jaxpr, out_pvals, consts = pe.trace_to_jaxpr(jvpfun_flat, in_pvals)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/partial_eval.py", line 513, in trace_to_jaxpr
    jaxpr, (out_pvals, consts, env) = fun.call_wrapped(pvals)
  File "/usr/local/lib/python3.7/dist-packages/jax/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/content/drive/MyDrive/Final Project/new_clrs/clrs/_src/baselines.py", line 947, in loss
    truth.data[i + 1] * jax.nn.log_softmax(
  File "/usr/local/lib/python3.7/dist-packages/jax/_src/numpy/lax_numpy.py", line 2628, in mean
    where=where)
  File "/usr/local/lib/python3.7/dist-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/jax/_src/api.py", line 419, in cache_miss
    donated_invars=donated_invars, inline=inline)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1632, in bind
    return call_bind(self, fun, *args, **params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1623, in call_bind
    outs = primitive.process(top_trace, fun, tracers, params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1635, in process
    return trace.process_call(self, fun, tracers, params)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/ad.py", line 323, in process_call
    result = call_primitive.bind(f_jvp, *primals, *nonzero_tangents, **new_params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1632, in bind
    return call_bind(self, fun, *args, **params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1623, in call_bind
    outs = primitive.process(top_trace, fun, tracers, params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1635, in process
    return trace.process_call(self, fun, tracers, params)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/partial_eval.py", line 203, in process_call
    f, in_pvals, app, instantiate=False)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/partial_eval.py", line 311, in partial_eval
    out_flat, (out_avals, jaxpr, env) = app(f, *in_consts), aux()
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1632, in bind
    return call_bind(self, fun, *args, **params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1623, in call_bind
    outs = primitive.process(top_trace, fun, tracers, params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 1635, in process
    return trace.process_call(self, fun, tracers, params)
  File "/usr/local/lib/python3.7/dist-packages/jax/core.py", line 627, in process_call
    return primitive.impl(f, *tracers, **params)
  File "/usr/local/lib/python3.7/dist-packages/jax/interpreters/xla.py", line 688, in _xla_call_impl
    *unsafe_map(arg_spec, args))
  File "/usr/local/lib/python3.7/dist-packages/jax/linear_util.py", line 261, in memoized_fun
    fun.populate_stores(stores)
  File "/usr/local/lib/python3.7/dist-packages/jax/linear_util.py", line 148, in populate_stores
    for self_store, other_store in zip(self.stores, stores):
KeyboardInterrupt